src/EditorFeatures/Test2/Simplification/CastSimplificationTests.vb

```diff
@@ -5035,6 +5035,89 @@ class C
             Await TestAsync(input, expected)
         End Function
 
+        <Fact, Trait(Traits.Feature, Traits.Features.Simplification)>
+        Public Async Function TestCSharp_DoNotSimplifyNullable() As Task
```

Comment:
I will admit this one might need a comment as to why it's the right behavior... :smile:

Reply:
(test name might also need some fancier name to explain why this is special)